generator/spring-boot/src/main/java/io/fabric8/maven/generator/springboot/SpringBootGenerator.java

```diff
@@ -71,14 +80,29 @@ public boolean isApplicable(List<ImageConfiguration> configs) {
 
     @Override
     public List<ImageConfiguration> customize(List<ImageConfiguration> configs, boolean isPrePackagePhase) throws MojoExecutionException {
-        if (!isPrePackagePhase && getContext().isWatchMode()) {
+        if (getContext().isWatchMode()) {
             ensureSpringDevToolSecretToken();
-            addDevToolsToFatJar(configs);
+            if (!isPrePackagePhase ) {
+                addDevToolsFilesToFatJar(configs);
+            }
         }
         return super.customize(configs, isPrePackagePhase);
     }
 
     @Override
+    protected Map<String, String> getEnv(boolean prePackagePhase) throws MojoExecutionException {
+        Map<String, String> res = super.getEnv(prePackagePhase);
+        if (getContext().isWatchMode()) {
```

Comment:
Yeah.. the watch is now a standalone task, so it creates an ad hoc build. In this case I think it's ok to produce different artifacts than in normal build, but, sure we should try to remove the goal finder.. or at least move the goal finder logic out of the core..